cmd/bio-rd/main.go

```diff
@@ -86,19 +118,60 @@ func main() {
 }
 
 func installSignalHandler() {
-	signal.Notify(sigHUP, syscall.SIGHUP)
+	signal.Notify(sigCh, syscall.SIGHUP)
+	signal.Notify(sigCh, syscall.SIGINT)
 }
 
-func configReloader() {
-	for {
-		<-sigHUP
-		log.Infof("Reloading configuration")
+func signalChecker() {
+	for sig := range sigCh {
+		log.Infof("Loading configuration")
 		newCfg, err := config.GetConfig(*configFilePath)
 		if err != nil {
 			log.Errorf("Failed to get config: %v", err)
 			continue
 		}
 
+		if sig != syscall.SIGHUP {
+			log.Infof("Received signal to STOP")
```

Comment:
This log message is fake news. A signal != syscall.SIGHUP was received. Not necessarily a SIGSTOP.